File: app_manager/appmanager/src/main/java/app/editors/manager/ui/views/pager/PagingViewPager.java
package app.editors.manager.ui.views.pager;


import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.viewpager.widget.ViewPager;

public class PagingViewPager extends ViewPager {

    public static final String TAG = PagingViewPager.class.getSimpleName();

    public interface OnPagerListener {
        void onPageScroll(boolean isActive);
        void onStartScroll();
    }

    private boolean mIsPaging = true;

    public PagingViewPager(Context context) {
        super(context);
    }

    public PagingViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        return mIsPaging && super.onTouchEvent(event);
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event) {
        try {
            return mIsPaging && super.onInterceptTouchEvent(event);
        } catch (IllegalArgumentException ex) {
            return false;
        }
    }

    public void setPaging(boolean isPaging) {
        mIsPaging = isPaging;
    }

    public boolean isPaging() {
        return mIsPaging;
    }

}

File: toolkit/libtoolkit/src/main/java/lib/toolkit/base/ui/views/pager/PagingViewPager.kt
package lib.toolkit.base.ui.views.pager


import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent

import androidx.viewpager.widget.ViewPager

class PagingViewPager : ViewPager {

    companion object {
        val TAG = PagingViewPager::class.java.simpleName!!
    }

    var isPaging = true

    interface OnPagerListener {
        fun onPageScroll(isActive: Boolean)
        fun onStartScroll()
    }

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    override fun onTouchEvent(event: MotionEvent): Boolean {
        return isPaging && super.onTouchEvent(event)
    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        return try {
            isPaging && super.onInterceptTouchEvent(event)
        } catch (ex: IllegalArgumentException) {
            false
        }
    }

}